packages/push/examples/median/main.rs

```diff
@@ -0,0 +1,210 @@
+pub mod args;
+
+use anyhow::Result;
+use clap::Parser;
+use ec_core::{
+    distributions::{collection::ConvertToCollectionGenerator, conversion::IntoDistribution},
+    generation::Generation,
+    individual::{ec::WithScorer, scorer::FnScorer},
+    operator::{
+        genome_extractor::GenomeExtractor,
+        genome_scorer::GenomeScorer,
+        mutator::Mutate,
+        selector::{best::Best, lexicase::Lexicase, Select, Selector},
+        Composable,
+    },
+    test_results::{self, TestResults},
+};
+use ec_linear::mutator::umad::Umad;
+use push::{
+    evaluation::cases::{Case, Cases, WithTargetFn},
+    genome::plushy::{GeneGenerator, Plushy},
+    instruction::{
+        variable_name::VariableName, BoolInstruction, ExecInstruction, IntInstruction,
+        PushInstruction,
+    },
+    push_vm::{program::PushProgram, push_state::PushState, stack::StackError, HasStack, State},
+};
+use rand::{
+    distributions::{Distribution, Uniform},
+    thread_rng, Rng,
+};
+use strum::IntoEnumIterator;
+
+use crate::args::{CliArgs, RunModel};
+
+#[derive(Copy, Clone)]
+struct Input([i64; 3]);
+
+#[derive(Copy, Clone)]
+struct Output(i64);
+
+impl Input {
+    fn median(&self) -> Output {
+        let Self(mut input) = self;
+        input.sort_unstable();
+        Output(input[1])
+    }
+}
+
+impl Distribution<Input> for Uniform<i64> {
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> Input {
+        Input([self.sample(rng), self.sample(rng), self.sample(rng)])
```

Comment:
This is probably nicer with
```rs
std::array::from_fn(|| self.sample(rng))
```
( didn't know that existed until a while ago, should probably do the same change for smallest )